src/Java.Interop.Tools.JavaSource/Java.Interop.Tools.JavaSource/SourceJavadocToXmldocGrammar.BlockTagsBnfTerms.cs

```diff
@@ -114,10 +114,23 @@ internal void CreateRules (SourceJavadocToXmldocGrammar grammar)
 					if (!grammar.ShouldImport (ImportJavadoc.ReturnTag)) {
 						return;
 					}
-					var r = new XElement ("returns",
+					// When encountering multiple @return keys in a line, append subsequent @return key content to the original <returns> element.
+					var jdi = FinishParse (context, parseNode);
+					if (jdi.Returns.Count == 0) {
+						var r = new XElement ("returns",
 							AstNodeToXmlContent (parseNode.ChildNodes [1]));
-					FinishParse (context, parseNode).Returns.Add (r);
-					parseNode.AstNode   = r;
+						FinishParse (context, parseNode).Returns.Add (r);
+						parseNode.AstNode = r;
+
+					} else {
+						var r = jdi.Returns.First () as XElement;
+						if (r != null) {
```

Comment:
This can't be `null`; `.First()` *requires* that *an* element be returned, and unless we have a bug and we do `.Returns.Add(null)`, there should only be non-`null` elements within `jdi.Returns`.

If we want to *allow* `null`, use `.FirstOrDefault()`.

Actually, on review this entire block doesn't *quite* make sense to me: `r` *itself* shouldn't change from `r.Add()`, and thus this *should* be a no-op (unless `jdi.Returns` somehow has > 1 element in it?  But how could that happen?)

```csharp
FinishParse (context, parseNode).Returns.Clear ();
FinishParse (context, parseNode).Returns.Add (r);
```

Neither of the above statements should be needed.

Reply:
The null check was intended to handle the possibility of `jdi.Returns.First ()` not being an `XElement`, though I think we only ever add `XElement`s to this collection in the section above and so the check probably shouldn't be needed.

The `FinishParse (context, parseNode).Returns` collection will have more than zero elements in it when the `<javadoc>` entry that is being parsed has multiple `@return` values in it, as is shown in the test addition below. The `Returns.Clear` and `Returns.Add` do look to be redundant here though, I will clean that up.